bitcoin/src/amount.rs

```diff
@@ -14,6 +14,21 @@ use core::{default, ops};
 use crate::prelude::*;
 
 /// A set of denominations in which amounts can be expressed.
+///
+/// # Examples
+/// ```
+/// # use core::str::FromStr;
+/// # use bitcoin::Amount;
+///
+/// assert_eq!(Amount::from_str("1 BTC").unwrap(), Amount::from_sat(100_000_000));
+/// assert_eq!(Amount::from_str("1 mBTC").unwrap(), Amount::from_sat(100_000));
+/// assert_eq!(Amount::from_str("1 uBTC").unwrap(), Amount::from_sat(100));
+/// assert_eq!(Amount::from_str("10 nBTC").unwrap(), Amount::from_sat(1));
+/// assert_eq!(Amount::from_str("10000 pBTC").unwrap(), Amount::from_sat(1));
+/// assert_eq!(Amount::from_str("1 bit").unwrap(), Amount::from_sat(100));
+/// assert_eq!(Amount::from_str("1 sat").unwrap(), Amount::from_sat(1));
+/// assert_eq!(Amount::from_str("1000 msats").unwrap(), Amount::from_sat(1));
```

Comment:
AFAIK `"1 BTC".parse::<Amount>()` is considered idiomatic rather than `FromStr` (which isn't even in the prelude) but doesn't matter much.

Reply:
Ah, I've always used `FromStr` and often wondered why it isn't in prelude when parsing strings is so common, TIL.

Reply:
Perhaps that comes from tuna-fish syntax, or what ever its called, being so difficult to me when I was a novice Rust hacker.